javascript/javascript3/week3/shoppingCart/shoppingCart.js

```diff
@@ -0,0 +1,166 @@
+
+            const currencyInArray = Object.entries(data); //converts objects into array so we can get every items key value pair in array.The Object.entries() method returns an array of arrays. fx. console.log(Object.entries(obj));
+            // Expected output: [["name", "Daniel"], ["age", 40],
+
+            currencyInArray.map((value) => {
+                if (value.includes(currency.toUpperCase())) {
+                    //console.log(value[0],value[1]);
+                    const convertedPrice = (this.price * value[1]).toFixed(2);
+                    console.log(convertedPrice + ' ' + value[0]);
+                }
+            });
+        } catch (error) {
+            console.log(error);
+        }
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        // Implement functionality here
+        this.products.push(product);
+    }
+
+    /*   findProduct(product) {
+        this.products.filter((product) => {
+            if (product.name === product) {
+                return `Product Name : ${product.name}
+              Price        : ${product.price}`;
+            }
+        });
+    } */
+
+    removeProduct(product) {
+        // Implement functionality here
+        const index = this.products.map((item) => item.name).indexOf(product);
+        this.products.splice(index, 1);
+    }
+
+    searchProduct(productName) {
+        // Implement functionality here
+        //first method
+        const search = this.products.map((product) => product.name);
+        if (search.includes(productName)) {
+            return productName; //+ " found";
+        } else {
+            return 'No matches';
+        }
+        /* const search = this.products.map((product) => product).filter((val)=>val.name.includes(productName));
+        console.log(search); */
+    }
+
+    getTotal() {
+        // Implement functionality here
+        console.log(this.products);
+        const total = this.products
+            .map((product) => product.price)
+            .reduce((total, currentPrice) => total + currentPrice);
+        return total;
+    }
+
+    renderProducts() {
+        // Implement functionality here
+        const pValue1 = document.querySelector('.product-name');
+        const pValue2 = document.querySelector('.price');
+        const pValue3 = document.querySelector('.total');
+
+        const searchItem = document.querySelector('.search-item');
+        searchItem.addEventListener('keyup', () => {
+            const val = searchItem.value;
+            const result = this.searchProduct(val);
+            if (result.includes('No matches')) {
+                pValue2.innerHTML = '';
+                pValue3.innerHTML = '';
+            } else {
+                pValue1.innerHTML = '';
+                pValue2.innerHTML = '';
+                pValue3.innerHTML = '';
+                var btn = document.createElement('button');
```

Comment:
const, and use meaningful name

Reply:
I wrote const in line 81 for pValue, do i need to write const again?? and i change to meaningful value. thanks